src/main/java/org/icatproject/lucene/Lucene.java

```diff
@@ -295,22 +417,22 @@ private void add(HttpServletRequest request, String entityName, When when, JsonP
 	@Path("addNow/{entityName}")
 	public void addNow(@Context HttpServletRequest request, @PathParam("entityName") String entityName)
 			throws LuceneException {
+		List<JsonObject> documents;
+		JsonStructure value = null;
 		logger.debug("Requesting addNow of {}", entityName);
-		int count = 0;
-		try (JsonParser parser = Json.createParser(request.getInputStream())) {
-			Event ev = parser.next(); // Opening [
-			while (true) {
-				ev = parser.next(); // Final ] or another document
-				if (ev == Event.END_ARRAY) {
-					break;
-				}
-				add(request, entityName, When.Now, parser, null);
-				count++;
+		try (JsonReader reader = Json.createReader(request.getInputStream())) {
+			value = reader.read();
+			documents = ((JsonArray) value).getValuesAs(JsonObject.class);
+			for (JsonObject document : documents) {
+				createNow(entityName, document);
 			}
+		} catch (JsonException e) {
+			logger.error("Could not parse JSON from {}", value.toString());
```

Comment:
To be honest I'm not sure. The fact that I've put a logger in there rather than just throwing means it probably did happen when I was developing (and I wanted extra logging to debug it). It's plausible it could happen when we `read()` if the input wasn't a valid `JsonStructure`, or later on when getting the values as Objects. In order to avoid the `NullPointer` could just remove the `toString` like elsewhere, and just let the logger format it?

Reply:
I think I added this comment before noticing the redundant usage of `toString` so I forgot to go back to it and update it, sorry. You are right, I believe that removing the `toString` should address this.